ios/FluentUI.Demo/FluentUI.Demo/Demos/DrawerVnextDemoController.swift

```diff
@@ -0,0 +1,145 @@
+            ],
+            itemSpacing: Constants.verticalSpacing,
+            stretchItems: true
+        )
+        addDescription(text: "Swipe from the left or right edge of the screen to reveal a drawer interactively")
+
+        container.addArrangedSubview(UIView())
+
+        // Screen edge gestures to interactively present side drawers
+
+        let leadingEdgeGesture = UIScreenEdgePanGestureRecognizer(target: self, action: #selector(handleScreenEdgePan))
+        leadingEdgeGesture.edges = view.effectiveUserInterfaceLayoutDirection == .leftToRight ? .left : .right
+        view.addGestureRecognizer(leadingEdgeGesture)
+        navigationController?.navigationController?.interactivePopGestureRecognizer?.require(toFail: leadingEdgeGesture)
+
+        let trailingEdgeGesture = UIScreenEdgePanGestureRecognizer(target: self, action: #selector(handleScreenEdgePan))
+        trailingEdgeGesture.edges = view.effectiveUserInterfaceLayoutDirection == .leftToRight ? .right : .left
+        view.addGestureRecognizer(trailingEdgeGesture)
+
+        drawerController = DrawerVnext(contentViewController: DrawerContentController())
+    }
+
+    @objc private func showLeftDrawerClearBackgroundButtonTapped() {
+        if let drawerController = drawerController {
+            drawerController.drawerState.backgroundDimmed = false
+            drawerController.drawerState.presentationDirection = .left
+            present(drawerController, animated: true, completion: nil)
+        }
+    }
+
+    @objc private func showLeftDrawerDimmedBackgroundButtonTapped() {
+        if let drawerController = drawerController {
+            drawerController.drawerState.backgroundDimmed = true
+            drawerController.drawerState.presentationDirection = .left
+            present(drawerController, animated: true, completion: nil)
+        }
+    }
+
+    @objc private func showRightDrawerClearBackgroundButtonTapped() {
+        if let drawerController = drawerController {
+            drawerController.drawerState.backgroundDimmed = false
+            drawerController.drawerState.presentationDirection = .right
+            present(drawerController, animated: true, completion: nil)
+        }
+    }
+
+    @objc private func showRightDrawerDimmedBackgroundButtonTapped() {
+        if let drawerController = drawerController {
+            drawerController.drawerState.backgroundDimmed = true
+            drawerController.drawerState.presentationDirection = .right
+            present(drawerController, animated: true, completion: nil)
+        }
+    }
+
+    @objc private func handleScreenEdgePan(gesture: UIScreenEdgePanGestureRecognizer) {
+        guard gesture.state == .began else {
+            return
+        }
+
+        let leadingEdge: UIRectEdge = view.effectiveUserInterfaceLayoutDirection == .leftToRight ? .left : .right
```

Comment:
hmmmm... I think the client app shouldn't need to worry about this.
Isn't there a way to just attach the drawerController to a specific view/viewcontroller and let it read the gestures and take the appropriate actions?

Reply:
I have created a new issue to track this 

https://github.com/microsoft/fluentui-apple/issues/356

Reply:
I have added a comment referencing the code of the current DrawerController to clarify that this logic should live in the control instead of the client app.